docs/source/reference/dataset_formats.rst

```diff
@@ -231,3 +444,147 @@ There are no shifts, missing hours, or extra hours for daylight savings time.
+    |    01001|   com|  warehouse|cooling |      2020|        2018|
+    |    01003|   com|large_hotel|heating |      2020|        2018|
+    +---------+------+-----------+--------+----------+------------+
+
+You can also use a simplified format with only the columns that vary::
+
+    +---------+-----------+
+    |geography|  subsector|
+    +---------+-----------+
+    |    01001|large_hotel|
+    |    01001|  warehouse|
+    |    01003|large_hotel|
+    +---------+-----------+
+
+Directory Format
+~~~~~~~~~~~~~~~~
+When using a directory, create separate files for different dimension combinations. File names
+should follow the pattern ``<dimension_type1>__<dimension_type2>.csv`` (note the double
+underscore).
+
+Example directory structure::
+
+    missing_associations/
+    ├── geography__subsector.csv
+    ├── geography__metric.csv
+    └── subsector__metric.csv
+
+Each file contains the relevant dimension columns::
+
+    # geography__subsector.csv
+    geography,subsector
+    01001,large_hotel
+    01001,warehouse
+
+Iterative Workflow for Identifying Missing Associations
+-------------------------------------------------------
+If you don't know which dimension combinations are missing in your dataset, dsgrid provides
+an iterative workflow to help you identify them:
+
+1. **Run registration without missing associations**
+
+   Attempt to register your dataset without specifying ``missing_associations``. If there are
+   missing combinations, registration will fail.
+
+2. **Review the generated output file**
+
+   When registration fails due to missing associations, dsgrid writes a Parquet file named
+   ``<dataset_id>__missing_dimension_record_combinations.parquet`` to the current directory.
+   This file contains all the missing dimension combinations with all dimensions. This file can
+   contain huge numbers of rows.
+
+3. **Analyze patterns in the missing data**
+
+   dsgrid also analyzes the missing data to identify minimal patterns that explain the gaps.
+   These patterns are logged and can help you understand *why* data is missing. For example,
+   you might see::
+
+       Pattern 1: geography | subsector = 01001 | large_hotel (150 missing rows)
+       Pattern 2: subsector = warehouse (3000 missing rows)
+
+   This tells you that all combinations involving county 01001 and large_hotel are missing,
+   and all combinations involving warehouse are missing.
+
+   dsgrid records these minimal patterns in dimension-specific combination files, such as
+   ``./missing_associations/geography__subsector.csv`` and
+   ``./missing_associations/sector__subsector.csv``.
+
+4. **Review and edit the missing associations file**
+
+   Examine the generated CSV files to verify that these combinations are legitimately missing
+   (not data errors). You may want to:
+
+   - Keep the files as they are if all missing combinations are expected.
+   - Remove rows that represent data errors you need to fix.
```

Comment:
This seems to need more editing. I think the user would review the generated output file and the missing data patterns at the same time and then decide how to proceed. I think they'd want to keep the missing data patterns that make sense to them and then rerun, eventually keeping some pattern files and a final missing-combinations file (if there are also one-off missings that are expected)?